.github/workflows/staging.yml

```
@@ -0,0 +1,106 @@
+name: Staging Build
+
+on:
+  push:
+    branches:
+      [staging-CI]
+
+jobs:
+  get-tag:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Get tagname of push
+        id: check
+        run: |
+          SHA=$(git log -n1 --format=format:"%H")
+          git checkout master
+          git describe $SHA
+          echo "The tag is " $(git describe $SHA)
+          echo ::set-output name=tagname::$(git describe $SHA) # change to exact-match 
```

```suggestion
          git describe --exact-match $SHA
          echo "The tag is " $(git describe --exact-match $SHA)
          echo ::set-output name=tagname::$(git describe --exact-match $SHA)
```
Exact match will fail if the currently pushed head is not at some tag. if we want to search for the last tag before the commit then we can use `--abbrev=0`